component-classes/index.js

```diff
@@ -220,41 +220,144 @@ export const expandable = {
+  utility: 'i-text-$color-button-utility-text hover:i-text-$color-button-utility-text i-bg-$color-button-utility-background i-border-$color-button-utility-border hover:i-bg-$color-button-utility-background hover:i-border-$color-button-utility-border-hover active:i-border-$color-button-utility-border-active',
+  destructive: 'i-bg-$color-button-negative-background i-text-$color-button-negative-text hover:i-text-$color-button-negative-text hover:i-bg-$color-button-negative-background-hover active:i-bg-$color-button-negative-background-active',
+  pill: 'i-text-$color-button-pill-icon hover:i-text-$color-button-pill-icon-hover active:i-text-$color-button-pill-icon-active i-bg-$color-button-pill-background hover:i-bg-$color-button-pill-background-hover active:i-bg-$color-button-pill-background-active',
+  disabled: 'i-text-$color-button-disabled-text i-bg-$color-button-disabled-background',
+  quiet: 'i-bg-$color-button-quiet-background i-text-$color-button-quiet-text hover:i-bg-$color-button-quiet-background-hover active:i-bg-$color-button-quiet-background-active',
+  utilityQuiet: 'i-text-$color-button-utility-quiet-text i-bg-$color-button-utility-quiet-background hover:i-bg-$color-button-utility-quiet-background-hover',
+  negativeQuiet: 'i-bg-$color-button-negative-quiet-background i-text-$color-button-negative-quiet-text hover:i-bg-$color-button-negative-quiet-background-hover active:i-bg-$color-button-negative-quiet-background-active',
+  loading: 'i-text-$color-button-loading-text i-bg-$color-button-loading-background',
+  link: 'i-text-$color-button-link-text',
+};
+
+const buttonTypes = {
+  primary: `border-0 rounded-8 ${buttonDefaultStyling}`,
+  secondary: `border-2 rounded-8 ${buttonDefaultStyling}`,
+  utility: `border rounded-4 ${buttonDefaultStyling}`,
+  negative: `border-0 rounded-8 ${buttonDefaultStyling}`,
+  pill:
+  `p-4 rounded-full border-0 inline-flex items-center justify-center hover:bg-clip-padding ${buttonDefaultStyling}`,
+  link: `max-w-max bg-transparent focusable ease-in-out inline active:underline hover:underline ${buttonColors.link}`,
+};
+
+const buttonSizes = {
+  xsmall: 'py-6 px-16',
+  small: 'py-8 px-16',
+  medium: 'py-10 px-14',
+  large: 'py-12 px-16',
+  utility: 'py-[11px] px-[15px]',
+  smallUtility: 'py-[7px] px-[15px]',
+  pill: 'min-h-[44px] min-w-[44px]',
+  pillSmall: 'min-h-32 min-w-32',
+  linkSmall: 'p-0',
+};
+
+const buttonTextSizes = {
+  medium: 'text-m leading-[24]',
+  xsmall: 'text-xs',
+};
+
+const buttonVariants = {
+  inProgress:
+    `border-transparent animate-inprogress pointer-events-none ${buttonColors.loading}`, // .button--in-progress, a.button--in-progress:visited
+  quiet:
+    `border-0 rounded-8 ${buttonDefaultStyling}`,
+  utilityQuiet: `bg-transparent border-0 rounded-4 ${buttonDefaultStyling}`,
```

Comment:
I'm confused that we have `bg-transparent` on `buttonVariants.utilityQuiet` and at the same time we have different colors in `buttonColors.utilityQuiet`. Is that correct?

Reply:
Good point, `bg-transparent` should be removed. I fix

Reply:
https://github.com/warp-ds/css/pull/44/commits/4a9940a0d61676c0e00b20107241998517bf41bb